src/System.Runtime/tests/System/String.SplitTests.cs

```diff
@@ -516,4 +496,22 @@ private static string[] ToStringArray(char[] source)
             return result;
         }
     }
+
+    // The below extension methods allow the tests to run on targets that *do not* have the new Split overloads.
+    // On targets that *do* have the new Split overloads (e.g. netcoreapp1.1), the actual overloads on String
+    // are called instead of the extension methods below (due to the way extension methods work).
+    internal static class StringSplitExtensions
```

Comment:
This seemed much simpler than splitting out the new overloads tests to their own `String.SplitTests.netcoreapp1.1.cs` file. I also considered creating a `StringSplitExtensions.cs` that is only included when `Condition="'$(TargetGroup)' != 'netcoreapp1.1'"`, but that seemed overly complex as well.

I confirmed that when running the tests with `/p:TargetGroup=netcoreapp1.1`, the actual `string.Split` methods are called instead of the extension methods.